pokemon.go

```diff
@@ -1,11 +1,110 @@
 package pokemonbattlelib
 
+import "fmt"
+
 type Pokemon struct {
-	// National Pokedex Number
-	NatDex uint16
-	// TODO
+    NatDex uint16                    // National Pokedex Number
+    Nickname string                  // player-given name for the Pokemon
+    Level Level                      // value from 1-100 influencing stats
+    Ability Ability                  // name of this Pokemon's ability
+    TotalExperience uint             // the total amount of exp this Pokemon has gained, influencing its level
+    Gender Gender                    // this Pokemon's gender
+    IVs [6]IndividualValue           // values from 0-31 that represents a Pokemon's 'genetic' potential
+    EVs [6]EffortValue               // values from 0-255 that represents a Pokemon's training in a particular stat
+    Nature Nature                    // represents a Pokemon's disposition and affects stats
+    Stats [6]Stat                    // the actual stats of a Pokemon determined from the above data
+    CurrentHP uint                   // the remaining HP of this Pokemon 
+    HeldItem Item                    // the item a Pokemon is holding
+    Moves [4]Move                    // the moves the Pokemon currenly knows
+    Friendship uint8                 // how close this Pokemon is to its Trainer
+    OriginalTrainerName string       // the name of the first Trainer who caught this Pokemon
+    OriginalTrainerID uint16         // a number associated with the first Trainer who caught this Pokemon
 }
 
 func (p *Pokemon) GetName() string {
 	return PokemonNames[p.NatDex]
 }
+
+/*
+func (p *Pokemon) GetExpGroup() ExpGroup {
+    //TODO
+}
+
+func (p *Pokemon) GetElementalTypes() (ElementalType, ElementalType) {
+    // TODO
+}
+
+func (p *Pokemon) GetBaseStats() [6]BaseStat {
+    // TODO
+} 
+
+func (p *Pokemon) GetMoveList() []Move {
+    //TODO
+}
+
+func (p *Pokemon) GetLearnSet() map[Level][]Move {
+    //TODO
+}
+
+func (p *Pokemon) GetBaseExperienceYield() uint8 {
+    //TODO
+} 
+
+func (p *Pokemon) GetEffortValueYield() [6]uint8 {
+    //TODO
+} 
+*/
+
+func (p *Pokemon) ComputeLevel() {
+    /*  TODO: Implement function to compute
+        pokemon's level based on its
+        total exp and exp group
+        and set the level to that computed amount
+    */
+} 
+
+func (p *Pokemon) ComputeStats() {
+    /* TODO: Implement function to compute
+       Pokemon's actual stats based on the Pokemon's
+       base stats, IVs, EVs, level, and Nature
+       and set the stats to the computed amounts
+    */
+}
```

Comment:
Remove these unimplemented functions. They assume too much about the implementation of what other people are working on